meetings/2018-01-17/minutes.md

```diff
@@ -0,0 +1,417 @@
+which holds the different components together.
+
+Eugene Yokata notes that sbt can provide a lot of details like type
+information. He notes there are two approaches, the SemanticDB and the
+presentation compiler, and asks where they can collaborate. He says that, as he
+sees it, the presentation compiler can work with incomplete data, but it can
+suffer from performance problems, and asks for clarification.
+
+In response, Stu Hood thinks that it's entirely possible for the SemanticDB to
+provide partial information. Regarding the "thick" vs "thin" model, he
+disagrees with Rory's suggestion of a bus, and suggests using a build tool as
+the "hub". He points out that the LSP server does not need a classpath.
+
+Jon sees the question more about whether we should standardize on a model, or
+whether this is something every build tool should define for itself. Stu
+suggests that it should be just "simple extensions" to the LSP protocol. Jon
+asks whether it might be simpler to treat the LSP and BSP as orthogonal
+projects, and Stu and Jorge both agree.
+
+Jorge and Stu debate some of the interactions between LSP and BSP. Stu notes
+that the LSP server does not need a compiler. The discussion goes into some
+technical detail, without reaching a conclusion. Stu would like LSP servers to
+be thinner.
+
+Jon tries with limited success to keep the conversation focussed away from
+technical details.
+
+Eugene Burmako added that he sees SemanticDB as a useful standard that works
+for both complete and partial information. He points out that it doesn't have
+to be persisted, so it can be exchanged in-memory. He notes that it's used to
+index third-party dependencies. It is used to communicate with the presentation
+compiler: the presentation compiler can produce SemanticDB data. Eugene
+suggests that SemanticDB could be a good standard for exchanging information
+between tools.
+
+Jon suggests that the LSP proposal is well-defined, and implementations exist,
+but suggests that the BSP should be designed to be modular to support more
+heterogeneous features. He suggest that different people could be involved with
+working on the two proposals.
+
+Shane Delmore asks about timescales for development, and asks what we're trying
+to support, and mentioned that people may want to work on projects which
+interoperate with other languages. Shane asks whether TASTY works currently
+with Scalac 2.11 or 2.12, and Adriaan says there's currently no schedule for
+including TASTY support in Scalac 2.13. He's open to it, but thinks it's more
+of a Scalac 2.14 target.
+
+Jon claims (in Martin's absence) that Martin didn't think it would be a huge
+amount of work to get a basic implementation of TASTY working for Scalac though
+challenges would exist in deciding the exact representations of Scalac
+structures in a format designed primarily with Dotty in mind, and Adriaan
+thinks that it's a challenge to get TASTY to scale between Java, Scala and
+Dotty.
+
```

Comment:
I had to leave before this part, and this may be an orthogonal point to the general discussion.  The is some discussion ongoing (https://github.com/twitter/reasonable-scala/issues/21) about using TASTY (or something similar format), shared backend protocol.  This would allow ScalaC/DotC/RSC to share a large component (code generation), make everybody's life better.  Its early days, but we are definitely exploring this area.

Reply:
Noted, though I'll leave this out of the minutes, as it might need quite a bit more explanation (which didn't happen in the meeting). :)